geps/gep-1897.md

```diff
@@ -0,0 +1,171 @@
+use case “As a client implementation of Gateway API, I need to know how to connect to
+a backend pod that has its own certificate”. TLS configuration can be a nebulous topic,
+so in order to drive resolution this GEP focuses only on this single piece of functionality.
+
+## Immediate Goals
+
+1. The solution must satisfy the following use case: the backend pod has its own
+certificate and the gateway implementation client needs to know how to connect to the
+backend pod. (Use case #4 in [Gateway API TLS Use Cases](#references))
+2. In terms of the Gateway API personas, only the application developer persona in this
+solution. The application developer should control the gateway to backend TLS settings,
+not the cluster operator, as requiring a cluster operator to manage certificate renewals
+and revocations would be extremely cumbersome.
+3. The solution should consider client certificate settings used in the TLS handshake **from
+Gateway to backend**, such as TLS versions and cipher suites. (Use case #5 in [Gateway API TLS Use Cases](#references))
+
+## Longer Term Goals
+
+These are worthy goals, but may need a different GEP for proper attention.
+
+1. TCPRoute use cases (completed by GA)
+2. Mutual TLS use cases
+3. Service mesh use cases
+
+## Non-Goals
+
+These are worthy goals, but will not be covered by this GEP.
+
+1. Changes to the existing mechanisms for edge or passthrough TLS termination
+2. Providing a mechanism to decorate multiple route instances
+3. TLSRoute use cases
+4. UDPRoute use cases 
+5. Termination of TLS for HTTP routing (#1 in [Gateway API TLS Use Cases](#references))
+6. HTTPS passthrough use cases (#2 in [Gateway API TLS Use Cases](#references))
+7. Termination of TLS for non-HTTP TCP streams (#3 in [Gateway API TLS Use Cases](#references))
+8. Controlling certificates used by more than one workload (#6 in [Gateway API TLS Use Cases](#references))
+9. Client certificate settings used in TLS **from external clients to the
+Listener** (#7 in [Gateway API TLS Use Cases](#references))
+10. Providing a mechanism for the cluster operator to override gateway to backend TLS settings.
+
+## Overview - what do we want to do?
+
+Given that the current ingress solution specifies **edge** TLS termination (from the client to
+the gateway), and how to handle **passthrough** TLS (from the client to the backend pod), this
+proposed ingress solution specifies TLS origination to the **backend** (from the gateway to the
+backend pod).  As mentioned, this solution satisfies the use case in which the backend pod
+has its own certificate and the gateway client needs to know how to connect to the backend pod.
+
+![image depicting TLS termination types](images/1897-TLStermtypes.png "TLS termination types")
+
+Gateway API is missing a mechanism for separately providing the details for the backend TLS handshake,
+including:
+* use of TLS
+* destination CA (certificate authority) or CA bundle
```

Comment:
Not sure if defining K8S root CAs is really in scope for this WG - my understanding was that there are separate WG working on certificate provisioning ( cert request, etc) and root CA configurations. 

Mutating/Validating webhooks are a major pain - I would be very careful around this UX, it is extremely hard to automate or use in helm charts.

Reply:
This is _not_ about defining K8S root CAs. Read the text.

This is about _being able to specify a CA bundle for the connection from the Gateway to the backend_. It's common that a service owner that is doing their own certificate management will manage their own root of trust, which will necessitate a separate CA bundle. It has nothing to do with Mutating or Validating webhooks.

Reply:
Indeed, but if you feel we've missed something and there's some changes to the language needed here please do drop a specific suggestion for change and/or let us know.